pydicom/config.py

```diff
@@ -152,15 +156,103 @@ def DS_decimal(use_Decimal_boolean: bool = True) -> None:
 Default ``False``.
 """
 
+
 enforce_valid_values = False
-"""Raise exceptions if any value is not allowed by DICOM Standard.
+"""Deprecated.
+Use :attr:`Settings.reading_validation_mode` instead.
+"""
 
-e.g. DS strings that are longer than 16 characters; IS strings outside
-the allowed range.
 
-Default ``False``.
+# Constants used to define how data element values shall be validated
+IGNORE = 0
+"""If one of the validation modes is set this value, no value validation
+will be performed.
+"""
+
+WARN = 1
+"""If one of the validation modes is set to this value, a warning is issued if
+a value validation error occurs.
 """
 
+RAISE = 2
+"""If one of the validation modes is set to this value, an exception is raised
+if a value validation error occurs.
+"""
+
+
+class Settings:
+    """Collection of several configuration values.
+    Accessed via the singleton :attr:`settings`.
+
+    .. versionadded:: 2.3
+    """
+
+    def __init__(self) -> None:
+        self._reading_validation_mode: Optional[int] = None
+        # in future version, writing invalid values will raise by default,
+        # currently the default value depends on enforce_valid_values
+        self._writing_validation_mode: Optional[int] = (
+            RAISE if _use_future else None
+        )
+
+    @property
+    def reading_validation_mode(self) -> int:
+        """Defines behavior of validation while reading values, compared with
+        the DICOM standard, e.g. that DS strings are not longer than
+        16 characters and contain only allowed characters.
+        The default (:attr:`WARN`) is to log a warning in the case of
```

Comment:
For WARN, does the warning log the warning or use `warnings.warn`?

Reply:
Use warnings.warn, mainly because this is what is used almost everywhere - but I wasn't sure about that.

Reply:
Updated the description.